apps/XMRig/install-32

```diff
@@ -0,0 +1,200 @@
+case $keepalived_conf in
+        Y|y|yes|Yes|YES) check=1; status_green "Your config now has keepalived packets ${inf_text}. You can change this by editing the '\"tls\":' field in $HOME/xmrig/bullseye/root/xmrig.json." ;;
+        N|n|no|No|NO) check=0 ;;
+	*) check=0; echo 'Invalid input! Please answer y/n, or yes/no.' ;;
+esac
+done
+
+# Set Wallet Address
+
+check=0
+until [ "$check" == "1" ]; do
+read -p "What is your Wallet Address? This will only be used in $HOME/xmrig/bullseye/root/.config/xmrig.json. " wallet_address
+read -p "Your Wallet Address will be set to $wallet_address. Is this ok? [Y/n] " wallet_conf
+case $wallet_conf in
+        Y|y|yes|Yes|YES) check=1; status_green "Your Wallet Address is now set to ${wallet_address}. You can change this by editing the '\"user\":' field in $HOME/xmrig/bullseye/root/xmrig.json." ;;
+        N|n|no|No|NO) check=0 ;;
+	*) check=0; echo 'Invalid input! Please answer y/n, or yes/no.' ;;
+esac
+done
+
+# Set miner name
+
+check=0
+read -p "Do you want to name your Miner? [Y/n] " qminername
+if [ "$qminername" == "y" ] || [ "$qminername" == "Y" ] || [ "$qminername" == "Yes" ] || [ "$qminername" == "yes" ] || [ "$qminername" == "YES" ]; then
+until [ "$check" == "1" ]; do
+read -p "What to you want to name your Miner? " miner_name
+read -p "Your Miner's name will be ${miner_name}. Is this ok? [Y/n] " name_conf
+case $name_conf in
+        Y|y|yes|Yes|YES) check=1; status_green "Your Miner's name is now set to ${miner_name}. You can change this by editing the '\"rig-id\":' field in $HOME/xmrig/bullseye/root/xmrig.json." ;;
+        N|n|no|No|NO) check=0 ;;
+	*) check=0; echo 'Invalid input! Please answer y/n, or yes/no.' ;;
+esac
+done
+elif [ "$qminername" == "n" ] || [ "$qminername" == "N" ] || [ "$qminername" == "No" ] || [ "$qminername" == "no" ] || [ "$qminername" == "NO" ]; then
+status "You chose not to name your Miner. If you change your mind, you can edit the '\"rig-id\":' line in '$HOME/xmrig/bullseye/root/xmrig.json'. Continuing..."
+fi
+
+# copy the settings to xmrig.conf
+
+cat << EOT | sudo tee $chroot_dir/root/.config/xmrig.json || error 'Failed to make $chroot_dir/root/xmrig.json!'
+{
+    "autosave": true,
+    "cpu": true,
+    "opencl": false,
+    "cuda": false,
+    "pools": [
+        {
+            "url": "${pool_pref}",
+            "user": "${wallet_address}",
+            "rig-id": "${miner_name}",
+            "keepalive": ${keepalived},
+            "tls": ${tls}
+        }
+    ]
+}
+EOT
+}
+
+# Hardware and kernel checks
+
+if grep -q ARMv6 /proc/cpuinfo; then
+  error "XMRig is not compatible with the Raspberry Pi Zero or other ARMv6 boards."
+fi
+
+if [ "$(uname -m)" != "aarch64" ] && [ "$(uname -m)" != "arm64" ]; then
+    while [ "$check" == "0" ]; do
+      read -n 3 -p 'You need a 64-bit kernel to install XMRig. Do you want me to enable 64-bit kernel? [Y/n] ' input
+      case $input in
+              Y|y|yes|Yes|YES) check=1; enable_64bit ;;
```

Comment:
This would also annoy users who need to enable the 64-bit kernel. Imaging: you get through setting up your miner, and after all that, Pi-Apps says you need to reboot and start completely over.
I'd suggest moving this to the beginning of the script.

Reply:
Like I said above, it is at the beginning. :smile: